### DISCUSSION TITLE
Should the standard library only expose algebraic effects?

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#9448] [Ideas]
The question is whether the standard library should only expose algebraic effects?

The upshot is that it keeps the library modular and more portable to new platforms. 

It also fits well with the fact that Flix has its own data types, e.g. Option and List are Flix things defined in Flix.

The downside is that programmers have to either use the algebraic effects (with their run/handle functions) or use Java interop.

But I think with our decent Java interop (which will only improve from here) that its probably worth it. I don't see why we should expose modules like e.g. File and Files in the future-- which are tied to a specific Java implementation behind the scenes.

**Comment by JonathanStarup** [comment]
I think its a good idea but it puts pressure on effects to be nice to use so we have to be on top of that
